docs/content/develop/test.md

```diff
@@ -0,0 +1,154 @@
+   }
+
+   resource "google_compute_network" "network" {
+     name                    = "my-network"
+     auto_create_subnetworks = false
+   }
+   ```
+3. Replace the id of the primary resource you are testing with `<%= ctx[:primary_resource_id] %>`.
+4. Replace fields that are identifiers, like `id` or `name`, with an appropriately named variable. For example, `<%= ctx[:vars]['subnetwork_name'] %>`. The resulting configuration for the above example would look like this:
+   ```tf
+   resource "google_compute_subnetwork" "<%= ctx[:primary_resource_id] %>" {
+     name          = "<%= ctx[:vars]['subnetwork_name'] %>"
+     ip_cidr_range = "10.1.0.0/16"
+     region        = "us-central1"
+     network       = google_compute_network.network.name
+   }
+
+   resource "google_compute_network" "network" {
+     name                    = "<%= ctx[:vars]['network_name'] %>"
+     auto_create_subnetworks = false
+   }
+   ```
+5. Modify the relevant `RESOURCE_NAME.yaml` file under [magic-modules/mmv1/products](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/products) to include an [`examples`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/provider/terraform/examples.rb) block with your test. For example:
+   ```yaml
+   examples:
+     - !ruby/object:Provider::Terraform::Examples
+       name: "compute_subnetwork_basic"
+       primary_resource_id: "example"
+       vars:
+         subnetwork_name: "example-subnet"
+         network_name: "example-network"
+   ```
+
+> **Warning:** Values in `vars` must include a `-` (or `_`). They [trigger the addition of a `tf-test` prefix](https://github.com/GoogleCloudPlatform/magic-modules/blob/6858338f013f5dc57729ec037883a3594441ea62/mmv1/provider/terraform/examples.rb#L244), which the sweeper uses to clean them up after tests run.
+6. If beta-only fields are being tested:
+   - Add `provider = google-beta` to every resource in the `*.tf.erb` config file.
+   - Add `min_version: beta` to the `examples` block in `RESOURCE_NAME.yaml`.
+{{< /tab >}}
+{{< tab "Handwritten" >}}
+This section assumes you've used the [Add a resource]({{< ref "/develop/resource.md" >}}) guide to create your handwritten resource, and you have a working MMv1 config.
+
+> **Note:** If not, you can create one now, or skip this guide and construct the test by hand. Writing tests by hand can sometimes be a better option if there is a similar test you can copy from.
+
+1. Add the test in MMv1. Repeat for all the standard tests you will need.
+2. [Generate the providers]({{< ref "/get-started/generate-providers.md" >}}).
+3. From the provider, copy and paste the generated `*_generated_test.go` file into [`magic-modules/mmv1/third_party/terraform/tests`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) as a new file call `*_test.go`.
+4. Using an editor of your choice, remove the `Example` suffix from all function names.
```

Comment:
👍 for merging into the following step.

I'm not sure what you mean about the "Add a resource" page, but I do think there is a pattern presently where `*Example` tests indicate that they are generated, and if our guide instructs users to leave the suffix, I think it will cause unnecessary confusion.

Reply:
I believe I was thinking of https://googlecloudplatform.github.io/magic-modules/develop/resource/#add-a-resource - the "Handwritten" tab here has very similar content related to generating tests and then copying the code.

Reply:
Ah, got it. Yea, I think those instructions as-written would leave the `Example` suffix on functions, and also leave the `generated_test` suffix in the filename, so I can add those pieces.

Reply:
Updated!